client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -247,43 +257,58 @@ public SendShuffleDataResult sendShuffleData(String appId, List<ShuffleBlockInfo
-              + commitCount + "], map number of stage is " + numMaps);
-          if (commitCount >= numMaps) {
-            RssFinishShuffleResponse rfsResponse =
-                getShuffleServerClient(ssi).finishShuffle(new RssFinishShuffleRequest(appId, shuffleId));
-            if (rfsResponse.getStatusCode() != ResponseStatusCode.SUCCESS) {
-              String msg = "Failed to finish shuffle to " + ssi + " for shuffleId[" + shuffleId
-                  + "] with statusCode " + rfsResponse.getStatusCode();
-              LOG.error(msg);
-              throw new Exception(msg);
-            } else {
-              LOG.info("Successfully finish shuffle to " + ssi + " for shuffleId[" + shuffleId + "]");
+    forkJoinPool.submit(() -> {
+      shuffleServerInfoSet.parallelStream().forEach(ssi -> {
+        RssSendCommitRequest request = new RssSendCommitRequest(appId, shuffleId);
+        String errorMsg = "Failed to commit shuffle data to " + ssi + " for shuffleId[" + shuffleId + "]";
+        long startTime = System.currentTimeMillis();
+        try {
+          RssSendCommitResponse response = getShuffleServerClient(ssi).sendCommit(request);
+          if (response.getStatusCode() == ResponseStatusCode.SUCCESS) {
+            int commitCount = response.getCommitCount();
+            LOG.info("Successfully sendCommit for appId[" + appId + "], shuffleId[" + shuffleId
+                + "] to ShuffleServer[" + ssi.getId() + "], cost "
+                + (System.currentTimeMillis() - startTime) + " ms, got committed maps["
+                + commitCount + "], map number of stage is " + numMaps);
+            if (commitCount >= numMaps) {
+              RssFinishShuffleResponse rfsResponse =
+                  getShuffleServerClient(ssi).finishShuffle(new RssFinishShuffleRequest(appId, shuffleId));
+              if (rfsResponse.getStatusCode() != ResponseStatusCode.SUCCESS) {
+                String msg = "Failed to finish shuffle to " + ssi + " for shuffleId[" + shuffleId
+                    + "] with statusCode " + rfsResponse.getStatusCode();
+                LOG.error(msg);
+                throw new Exception(msg);
+              } else {
+                LOG.info("Successfully finish shuffle to " + ssi + " for shuffleId[" + shuffleId + "]");
+              }
             }
+          } else {
+            String msg = errorMsg + " with statusCode " + response.getStatusCode();
+            LOG.error(msg);
+            throw new Exception(msg);
           }
-        } else {
-          String msg = errorMsg + " with statusCode " + response.getStatusCode();
-          LOG.error(msg);
-          throw new Exception(msg);
+          successfulCommit.incrementAndGet();
+        } catch (Exception e) {
+          LOG.error(errorMsg, e);
         }
-        successfulCommit.incrementAndGet();
-      } catch (Exception e) {
-        LOG.error(errorMsg, e);
-      }
-    });
+      });
+    }).join();
+    forkJoinPool.shutdownNow();
```

Comment:
We add method `shutdownNow`. Maybe we'd better do performance tests again.

Reply:
Enn. As i know, the shutDownNow actually will not effect the performance.

Reply:
Should we use 
```
try {

} finally {
  forkJoinPool.shutdownNow();
}
```

Reply:
https://stackoverflow.com/questions/70934558/forkjoinpoolshutdown-vs-forkjoinpoolshutdownnow-after-forkjoinpooljoin

Reply:
It looks ShutdownNow or Shutdown are all OK due to the operation of `join`

Reply:
> It looks ShutdownNow or Shutdown are all OK due to the operation of `join`
It's ok for me to use method `shutdownNow()`.

Reply:
I just think we can't shutdown forkjoinpool because of exception.

Reply:
Got it.